recipes/tree-gen/all/conanfile.py

```diff
@@ -62,20 +72,16 @@ def build_requirements(self):
                 self.tool_requires("flex/2.6.4")
                 self.tool_requires("bison/3.8.2")
 
-    def requirements(self):
-        self.requires("fmt/10.2.1")
-        self.requires("range-v3/0.12.0")
-
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
-
-    def configure(self):
-        if self.options.shared:
-            self.options.rm_safe("fPIC")
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.")
 
-    def layout(self):
-        cmake_layout(self, src_folder="src")
+    def requirements(self):
+        self.requires("fmt/10.2.1", transitive_headers=True)
```

Comment:
Hi @uilianries , can you explain what this `transitive_headers` do?

Reply:
In Conan 2.x all requires are private, including dependencies. Which means, when importing tree-gen recipe, fmt and range-v3 will not be available for who is consuming, so the consumer should be explicit about what wants. However, in case tree-gen headers export fmt or range-v3, the consumer will not be able to use package, unless those dependencies are public too. The transitive headers exports the dependency headers as public.

https://docs.conan.io/2/reference/conanfile/methods/requirements.html#transitive-headers

Reply:
Thanks so much for the explanation!